pkg/internal/diskutil.go

```diff
@@ -120,6 +120,29 @@ func (b BlockDevice) HasChildren() (bool, error) {
 	return false, nil
 }
 
+// HasBindMounts checks for bind mounts in targetDirectory using `findmnt` command
+// Returns true if targetDirectory (target) and device name (source) are present in the ouput
+func (b BlockDevice) HasBindMounts(targetDir string) (bool, error) {
+	args := []string{"--raw"}
+	cmd := ExecCommand("findmnt", args...)
+	output, err := executeCmdWithCombinedOutput(cmd)
+	if err != nil {
```

Comment:
Should we use `findmnt` or `cat /proc/1/mountinfo` ? I personally like using `/proc/1/mountinfo` because that means we depend on one less utility inside the pod.

Reply:
using `cat /proc/1/mountinfo` to get the bind mount details.

Reply:
Is there any reason to use `cat` and exec? can you not just open the file `/proc/1/mountinfo` and read it - rather than using exec interface?

But there are other problems here:
1. looking for a substring (such as `sda`) can match the mount path inadvertently.  according to - https://man7.org/linux/man-pages/man5/proc.5.html , we should only search for device name match in 4th and 10th field. 
2. If we correct device name matching as above, we don't need to know kubelet mount directory here. If a mount exists for the device then device can't be used.

Reply:
> Is there any reason to use `cat` and exec? can you not just open the file `/proc/1/mountinfo` and read it - rather than using exec interface?
Updated code to read contents of the file. 

> 1. looking for a substring (such as `sda`) can match the mount path inadvertently.  according to - https://man7.org/linux/man-pages/man5/proc.5.html , we should only search for device name match in 4th and 10th field (search for `mountinfo` format description in the linked doc)

Thanks for the pointers. I could make use of the 4th field to get the `root` and 5th field to get the `mountpoint`.  
```5595 121 0:6 **/sdc** **/var/lib/kubelet/plugins/kubernetes.io~local-volume/volumeDevices/local-pv-343bdd9/6d9d33ae-408e-4bac-81f7-c0bc347a9667** rw shared:23 - devtmpfs devtmpfs rw,seclabel,size=32180404k,nr_inodes=8045101,mode=755```